pkg/queue/concurrency_state_test.go

```diff
@@ -278,3 +323,27 @@ func BenchmarkConcurrencyStateProxyHandler(b *testing.B) {
 		reportTicker.Stop()
 	}
 }
+
+// createTempTokenFile creates a temporary file with the text "0123456789" for simulating a serviceAccountToken
+// Note that it does NOT delete the temp file, this must be called separately, for example:
+//
+// tempFile := createTempTokenFile(logger)
+// defer os.Remove(tempFile.Name())
+func createTempTokenFile(logger *zap.SugaredLogger) *os.File {
```

Comment:
Would t.TempDir work here? That would automatically clean up the dir when the test exits. (We should avoid ioutil package for new code, all the methods now have equivalents elsewhere).